main.go

```diff
@@ -124,118 +132,174 @@ func main() {
+	writeBadData(bw, conv, banner, outputFilePrefix+badDataFile, ioHelper.out)
+	return nil
+}
+
+func schemaConv(driver string, ioHelper *ioStreams) (*internal.Conv, error) {
+	switch driver {
+	case POSTGRES:
+		return schemaConvFromSourceDB(driver)
+	case PGDUMP:
+		return schemaConvFromFileStream(ioHelper)
+	default:
+		return nil, fmt.Errorf("schema conversion for driver %s not supported", driver)
+	}
+}
+
+func dataConv(driver string, ioHelper *ioStreams, client *sp.Client, conv *internal.Conv) (*spanner.BatchWriter, error) {
+	switch driver {
+	case POSTGRES:
+		return dataConvFromSourceDB(ioHelper, client, conv)
+	case PGDUMP:
+		return dataConvFromFileStream(ioHelper, client, conv)
+	default:
+		return nil, fmt.Errorf("data conversion for driver %s not supported", driver)
+	}
+}
+
+func driverConfig(driver string) (string, error) {
+	switch driver {
+	case POSTGRES:
+		return pgDriverConfig()
+	default:
+		return "", fmt.Errorf("Driver %s not supported", driver)
+	}
+}
+
+func pgDriverConfig() (string, error) {
 	server := os.Getenv("PGHOST")
 	port := os.Getenv("PGPORT")
 	user := os.Getenv("PGUSER")
 	dbname := os.Getenv("PGDATABASE")
 	if server == "" || port == "" || user == "" || dbname == "" {
 		fmt.Printf("Please specify host, port, user and database using PGHOST, PGPORT, PGUSER and PGDATABASE environment variables\n")
-		return fmt.Errorf("Could not connect to source database")
+		return "", fmt.Errorf("Could not connect to source database")
 	}
 	password := getPassword()
-	sourceDB, err := sql.Open(driver, fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", server, port, user, password, dbname))
+	return fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", server, port, user, password, dbname), nil
+}
+
+func schemaConvFromSourceDB(driver string) (*internal.Conv, error) {
+	fmt.Println(`###################################################################
+	Accessing a source DB via an database/sql driver is an experimental
+	feature that is not fully implemented. Currently we only do schema
+	conversion and we only support PostgreSQL.
+	###################################################################`)
+
+	driverConfig, err := driverConfig(driver)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	sourceDB, err := sql.Open(driver, driverConfig)
+	if err != nil {
+		return nil, err
 	}
 	conv := internal.MakeConv()
 	err = internal.ProcessInfoSchema(conv, sourceDB)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	writeSchemaFile(conv, now, outputFilePrefix+schemaFile)
-	spannerDB, err := createDatabase(projectID, instanceID, dbName, conv)
-	if err != nil {
-		fmt.Printf("\nCan't create database: %v\n", err)
-		return fmt.Errorf("can't create database")
+	return conv, nil
+}
+
+func dataConvFromSourceDB(ioHelper *ioStreams, client *sp.Client, conv *internal.Conv) (*spanner.BatchWriter, error) {
```

Comment:
I don't think the ioHelper would ever be needed here?
For data conversion, we only use ioHelper.seekableIn, and that should only happen when we're reading pgdump data i.e. dataConvFromFileStream.